src/JsonSchema/TypeFactory.php

```diff
@@ -50,14 +56,37 @@ public function setSchemaFactory(SchemaFactoryInterface $schemaFactory): void
     public function getType(Type $type, string $format = 'json', ?bool $readableLink = null, ?array $serializerContext = null, Schema $schema = null): array
     {
         if ($type->isCollection()) {
-            $subType = new Type($type->getBuiltinType(), $type->isNullable(), $type->getClassName(), false);
-
-            return [
-                'type' => 'array',
-                'items' => $this->getType($subType, $format, $readableLink, $serializerContext, $schema),
-            ];
+            $keyType = $type->getCollectionKeyType();
+            $subType = $type->getCollectionValueType()
```

Comment:
I think we prefer to have it in one line.

Reply:
I work on 3440x1440, and this was already quite too wide :D 

I don't mind reverting, but please focus review on non-cs issues first.

Reply:
There's word wrapping. Just saying...